org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandlerTest.java

```diff
@@ -183,6 +182,115 @@ public void testFormatting_onOffTags() throws Exception {
 
 	}
 
+	@Test
+	public void testFormatting_indent_switchstatementsDefault() throws Exception {
+		ICompilationUnit unit = getWorkingCopy("src/org/sample/Baz.java",
+		//@formatter:off
+			"package org.sample;\n" +
+			"\n" +
+			"public class Baz {\n" +
+			"    private enum Numbers {One, Two};\n" +
+			"    public void foo() {\n" +
+			"        Numbers n = Numbers.One;\n" +
+			"        switch (n) {\n" +
+			"        case One:\n" +
+			"        return;\n" +
+			"        case Two:\n" +
+			"        return;\n" +
+			"        default:\n" +
+			"        break;\n" +
+			"        }\n" +
+			"    }\n" +
+			"}"
+		//@formatter:off
+		);
+		String uri = JDTUtils.toURI(unit);
+		TextDocumentIdentifier textDocument = new TextDocumentIdentifier(uri);
+		FormattingOptions options = new FormattingOptions(4, true);
+		DocumentFormattingParams params = new DocumentFormattingParams(textDocument, options);
+		List<? extends TextEdit> edits = server.formatting(params).get();
+		assertNotNull(edits);
+		//@formatter:off
+		String expectedText =
+			"package org.sample;\n" +
+			"\n" +
+			"public class Baz {\n" +
+			"    private enum Numbers {\n" +
+			"        One, Two\n" +
+			"    };\n" +
+			"\n" +
+			"    public void foo() {\n" +
+			"        Numbers n = Numbers.One;\n" +
+			"        switch (n) {\n" +
+			"            case One:\n" +
+			"                return;\n" +
+			"            case Two:\n" +
+			"                return;\n" +
+			"            default:\n" +
+			"                break;\n" +
+			"        }\n" +
+			"    }\n" +
+			"}";
+		//@formatter:on
+		String newText = TextEditUtil.apply(unit, edits);
+		assertEquals(expectedText, newText);
+	}
+
+	@Test
+	public void testFormatting_indent_switchstatementsFalse() throws Exception {
+		javaProject.setOption(DefaultCodeFormatterConstants.FORMATTER_INDENT_SWITCHSTATEMENTS_COMPARE_TO_SWITCH, DefaultCodeFormatterConstants.FALSE);
```

Comment:
if this test runs 1st, javaProject's options will keep that setting when testFormatting_indent_switchstatementsDefault() runs

Reply:
Fixed.